### DISCUSSION TITLE
Does python-socketio's room work with multiple workers?

### DISCUSSION THREAD

**Discussion by RuidongLiu** [miguelgrinberg/python-socketio#1355] [General]
it seems in javascript, https://socket.io/docs/v4/rooms/#with-multiple-socketio-servers socketio will work with multiple workers. 

but for python socketio, rooms seems to be managed in memory. want to know if python socketio supports it.

**Comment by miguelgrinberg** [comment]
It does, yes. Multiple workers require the use of a message queue, which the workers use to coordinate operations such as emitting to a room.

**Reply by RuidongLiu** [reply]
nice! just random question: is it possible to get all clients in a room? or that information has to be stored in some database, like redis with multiple workers?

**Reply by miguelgrinberg** [reply]
You have to keep track of this information in your own database.
